src/prefect/orion/schemas/core.py

```diff
@@ -314,11 +318,30 @@ def populate_deprecated_fields(cls, values):
         """
         if not values.get("retries", None) and values.get("max_retries", 0) != 0:
             values["retries"] = values["max_retries"]
+
         if (
             not values.get("retry_delay", None)
             and values.get("retry_delay_seconds", 0) != 0
         ):
             values["retry_delay"] = values["retry_delay_seconds"]
+
+        return values
+
+    @root_validator
+    def validate_retry_and_jitter_inputs(cls, values):
+        """
+        Limits the number of configured retry delays to a maximum of 50. And ensures
+        that the relative jitter factor is positive.
+        """
+        if isinstance(values["retry_delay"], list) and (
+            len(values["retry_delay"]) > 50
+        ):
+            values["retry_delay"] = values["retry_delay"][:50]
```

Comment:
Do we want to silently ignore the extra items? I feel like we should raise an error?

Reply:
I could be convinced that we should have this truncate and just document it, but an error seems safer to start.

Reply:
I've noticed patterns where we'll "ignore" misconfigured thing in the past on the backend, I don't personally have a principled stance here

Reply:
Hm weird. I think we should avoid that wherever possible, maybe we'll need to audit some of the existing patterns.